modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/CompareVersionsJournalArticleContentDashboardItemVersionAction.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.content.dashboard.journal.internal.item.action;
+
+import com.liferay.content.dashboard.item.action.ContentDashboardItemVersionAction;
+import com.liferay.journal.constants.JournalPortletKeys;
+import com.liferay.journal.model.JournalArticle;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.portlet.LiferayPortletRequest;
+import com.liferay.portal.kernel.portlet.LiferayPortletResponse;
+import com.liferay.portal.kernel.portlet.url.builder.PortletURLBuilder;
+import com.liferay.portal.kernel.util.JavaConstants;
+import com.liferay.portal.kernel.util.PortalUtil;
+
+import java.util.Locale;
+
+import javax.portlet.PortletRequest;
+import javax.portlet.PortletResponse;
+import javax.portlet.PortletURL;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Jürgen Kappler
+ */
+public class CompareVersionsJournalArticleContentDashboardItemVersionAction
+	implements ContentDashboardItemVersionAction {
+
+	public CompareVersionsJournalArticleContentDashboardItemVersionAction(
+		HttpServletRequest httpServletRequest, JournalArticle journalArticle,
+		JournalArticle latestJournalArticle, Language language) {
+
+		_httpServletRequest = httpServletRequest;
+		_journalArticle = journalArticle;
+		_latestJournalArticle = latestJournalArticle;
+		_language = language;
+	}
+
+	@Override
+	public String getIcon() {
+		return null;
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return _language.get(locale, "compare-to");
+	}
+
+	@Override
+	public String getName() {
+		return "compare-versions";
+	}
+
+	@Override
+	public String getURL() {
+		LiferayPortletResponse liferayPortletResponse =
+			PortalUtil.getLiferayPortletResponse(
+				(PortletResponse)_httpServletRequest.getAttribute(
+					JavaConstants.JAVAX_PORTLET_RESPONSE));
+
+		LiferayPortletRequest liferayPortletRequest =
+			PortalUtil.getLiferayPortletRequest(
+				(PortletRequest)_httpServletRequest.getAttribute(
+					JavaConstants.JAVAX_PORTLET_REQUEST));
+
+		PortletURL portletURL = liferayPortletResponse.createRenderURL();
+
+		return PortletURLBuilder.create(
+			PortalUtil.getControlPanelPortletURL(
```

Comment:
@Stephy6, can we do something like this here?

```
_requestBackedPortletURLFactory.createRenderURL(
				JournalPortletKeys.JOURNAL) 
```

Reply:
Done! thanx!